src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -284,9 +287,24 @@ public String getPassword()
    @Override
    public void setPassword(String password)
    {
+      if (passwordSupplier != null) {
```

Comment:
If accepted maybe it would make sense to always use a supplier and just do
```java
this.passwordSupplier = () -> password;
```

Reply:
It seems a nice idea at first, but I think replacing the ``Supplier`` might be unexpected.  I think I prefer the explicit failure so the developer immediately understands they have an ambiguous configuration.